ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/PullQueryResponseHandler.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api.client.impl;
+
+import io.confluent.ksql.api.client.Row;
+import io.confluent.ksql.api.server.protocol.QueryResponseMetadata;
+import io.confluent.ksql.rest.client.KsqlRestClientException;
+import io.vertx.core.Context;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.parsetools.RecordParser;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+public class PullQueryResponseHandler extends QueryResponseHandler<List<Row>> {
+
+  private final List<Row> rows;
+  private List<String> columnNames;
+  private List<String> columnTypes;
+
+  PullQueryResponseHandler(final Context context, final RecordParser recordParser,
+      final CompletableFuture<List<Row>> cf) {
+    super(context, recordParser, cf);
+    this.rows = new ArrayList<>();
+  }
+
+  @Override
+  protected void handleMetadata(final QueryResponseMetadata queryResponseMetadata) {
+    if (queryResponseMetadata.queryId != null && !queryResponseMetadata.queryId.isEmpty()) {
+      cf.completeExceptionally(
+          new KsqlRestClientException("Expecting pull query but was push query"));
+    }
+
+    columnNames = queryResponseMetadata.columnNames;
+    columnTypes = queryResponseMetadata.columnTypes;
+  }
+
+  @Override
+  protected void handleRow(final Buffer buff) {
+    final JsonArray values = new JsonArray(buff);
+    rows.add(new RowImpl(columnNames, columnTypes, values));
```

Comment:
It might be worth adding a check here to make sure there aren't too many rows. E.g. somewhat might use this with a non terminating push query and it ends up exhausting all memory on the client. Once more than N rows are obtained throw an exception and abort.

Reply:
What do you think is a reasonable limit? 1000? I guess we could also make this configurable via the client options.

Reply:
+1 I think it should be configurable. 1000 rows seems a bit too few. How about 10000?

Reply:
Done. The config is currently a client config (rather than a param on the `executeQuery()` method itself) which makes more sense to me, though it does mean a user would have to client a new client rather than simply reissuing the request if they hit the limit and want to increase it.

The config name I put in (`setExecuteQueryMaxResultRows()`) isn't great. Open to suggestions!